src/components/app/MainContent.vue

```diff
@@ -3,68 +3,59 @@
     <div class="d-flex justify-center align-center">
       <v-btn
         x-large
-        :loading="scraping"
+        :loading="inProgress"
         :color="btnColor"
         @click="scrape"
       >
         Run
       </v-btn>
     </div>
-    <div>
-      <log-lines :entries="results" />
+    <div class="keep-bottom">
+      <log-viewer :entries="entries" />
     </div>
     <div>
       <config-editor />
     </div>
   </div>
 </template>
 
-<script>
+<script lang="ts">
+import LogViewer from '@/components/shared/log/LogViewer.vue';
+import Vue from 'vue';
+import { ref, computed } from '@vue/composition-api';
 import { scrapeAndUpdateOutputVendors } from '@/originalBudgetTrackingApp';
-import LogLines from '@/components/shared/LogLines';
-import ConfigEditor from './ConfigEditor';
+import ConfigEditor from './ConfigEditor.vue';
+import { LogEntry } from '../shared/log/types';
+import LogsEventEmitter from './LogsEventEmitter';
 
 const colors = {
-  true: 'green',
-  false: 'red',
-  null: null
+  success: 'green',
+  failed: 'red',
+  unknown: null
 };
 
-export default {
-  name: 'MainContent',
+export default Vue.extend({
   components: {
-    LogLines, ConfigEditor
+    LogViewer, ConfigEditor
   },
-  data() {
+  setup() {
+    const inProgress = ref(false);
+    const succeeded = ref('unknown' as keyof typeof colors);
```

Comment:
This doesn't ever change. Don't think this was what you intended.
Need to add back the logic that updates according to result of scraping

Reply:
In general I think we should replace succeeded and in progress with a single status. Something like:
status: NOT_STARTED | IN_PROGRESS | SUCCESS | FAILURE

Then colors becomes STATUS_COLORS: Record<ScrapingStatus, string | null>

Reply:
You can actually use the event emitter for setting the status instead of awaiting the promise (not a must):

eventPublisher.on(EventNames.IMPORT_PROCESS_START, () => {
    // IN_PROGRESS
  });
  
  eventPublisher.on(EventNames.IMPORT_PROCESS_END, () => {
    // SUCCESS
  });
  
  eventPublisher.on(EventNames.GENERAL_ERROR, () => {
    // ERROR
  });

Reply:
> In general I think we should replace succeeded and in progress with a single status. Something like:
> status: NOT_STARTED | IN_PROGRESS | SUCCESS | FAILURE
> 
> Then colors becomes STATUS_COLORS: Record<ScrapingStatus, string | null>

I think I don't understand... You just suggesting to change the `unknown | success | fail` with these `NOT_STARTED | IN_PROGRESS | SUCCESS | FAILURE`?

Reply:
> You can actually use the event emitter for setting the status instead of awaiting the promise (not a must):
> 
> eventPublisher.on(EventNames.IMPORT_PROCESS_START, () => {
> // IN_PROGRESS
> });
> 
> eventPublisher.on(EventNames.IMPORT_PROCESS_END, () => {
> // SUCCESS
> });
> 
> eventPublisher.on(EventNames.GENERAL_ERROR, () => {
> // ERROR
> });

As I said before, I don't like to use `EventEmitter` as a "serial" flow, because I think it is less readable.

Reply:
Moved this discussion to the phone. Easier to communicate that way 🙂

Reply:
Discussed on the phone. Easier to communicate that way 🙂

Reply:
I did it. I started with `enum` but I think for now we can just use the `keyof typeof statusToColor`